_posts/2020-05-13-#17681.md

```diff
@@ -34,7 +34,63 @@ commit:
+
+2. How does it help users to top up keypools for inactive HD seeds? What is not
+   good about the behavior before this PR?
+
+3. The original motivation for the PR was to remove the restriction that
+   setting a new HD seed wasn't [allowed during Initial Block
+   Download](https://github.com/bitcoin/bitcoin/commit/769b03a83c2aa2b97f344b58dc689be26c6e08e5)?
+   What was the reason for this restriction?
+
+4. Why was the `sethdseed` RPC added? Are there uses for having multiple HD
+   seeds in the same wallet? Does this change to `sethdseed` affect new
+   [descriptor
+   wallets](https://github.com/achow101/bitcoin/blob/desc-wallet-followup/doc/release-notes-16528.md)
+   as well as existing wallets?
+
+5. The main data structures used in this PR are
+   [`CHDChain`](https://github.com/ryanofsky/bitcoin/blob/review.17681.5/src/wallet/walletdb.h#L88),
+   [`CKeyMetadata`](https://github.com/ryanofsky/bitcoin/blob/review.17681.5/src/wallet/walletdb.h#L126),
+   and
+   [`KeyOriginInfo`](https://github.com/ryanofsky/bitcoin/blob/review.17681.5/src/script/keyorigin.h#L11).
+   What are the purposes of the different data structures?
+
+6. After the wallet has seen a existing key being used, how does the wallet
+   decide whether to "top up" and generate new keys? What factors determine
+   whether the wallet does or does not top up? Are there differences here
+   between the new
+   [`TopUpInactiveHDChain`](https://github.com/ryanofsky/bitcoin/blob/review.17681.5/src/wallet/scriptpubkeyman.cpp#L293)
+   method and existing
+   [`TopUp`](https://github.com/ryanofsky/bitcoin/blob/review.17681.5/src/wallet/scriptpubkeyman.cpp#L1194)
+   method?
+
+7. When is
+   [`AddInactiveHDChain`](https://github.com/ryanofsky/bitcoin/blob/review.17681.5/src/wallet/scriptpubkeyman.cpp#L902)
+   called? Where do the inactive hd chain objects come from? Is
+   `AddInactiveHDChain` called everywhere it needs to be called?
+
+8. How is the PR structured and divided up? Would it make sense to split up or combine commits?
+
+9. An [earlier
+   version](https://github.com/ryanofsky/bitcoin/commits/review.17681.1) of
+   this PR tried (the implementation was buggy) to store
+   [`m_inactive_hd_chains`](https://github.com/ryanofsky/bitcoin/blob/review.17681.1/src/wallet/scriptpubkeyman.h#L301)
+   map entries as
+   [`"inactivehdchain"`](https://github.com/ryanofsky/bitcoin/blob/review.17681.1/src/wallet/walletdb.cpp#L32)
+   rows in the wallet database. Was it good to stop doing this? What are advantages of
+   storing this information? What are the advantages of not storing it?
+
+10. A [previous
+   version](https://github.com/ryanofsky/bitcoin/commits/review.17681.4) or
+   this PR had a subtle bug on [this
+   line](https://github.com/ryanofsky/bitcoin/blob/review.17681.4/src/wallet/walletdb.cpp#L446).
+   What was the bug and what were the effects?
+
+11. Is test coverage in this PR good? Are there ways the test could be extended?
```

Comment:
For the first sentence, maybe: "Do you think this PR has sufficient test coverage?"